### ISSUE TITLE
Pet can't hit target if in the same position

### ISSUE THREAD

**Issue by merloechoed** [echoes-emu/echoes-bt#741]
Happened a few times during the open beta, but could trigger again easily using "!speed 100" command, as pet also gets the speed buff.
If the pet ends up on top of the target, it gets in a kind of stuck situation, that even if ordered to attack said target, the target can't be hit by it, as shown in video below.

Probably happens to all companions, like captain's herald.

Probably a "not facing target" issue.

https://user-images.githubusercontent.com/122577244/212241709-3ac2e07e-219d-4be0-95dc-eac16dea21e5.mp4

**Comment by woodx319** [comment]
how did your raven get in that spot in the first place? did you order to attack and it flew over there and that's where it was chilling?

**Comment by merloechoed** [comment]
> how did your raven get in that spot in the first place? did you order to attack and it flew over there and that's where it was chilling?

Exactly, it's just an "attack" command in the pet skills bar and sometimes, they end up in the exact position the target is, and the bug happens again.

Just triggered again in another mob:

![ScreenShot00065](https://user-images.githubusercontent.com/122577244/212245186-7349c381-e912-4f06-9cc1-66725261e087.jpg)

**Comment by merloechoed** [comment]
A better video demonstrating the issue:

https://user-images.githubusercontent.com/122577244/212246381-4ba29a98-1736-453e-8b1d-218aff35ceee.mp4

Using "!speed 100" command just to trigger it easily, but it happened in open beta quite a few times.

**Comment by merloechoed** [comment]
Looks like the way it works is when we use the pet attack command, the pet copies the target's position and rotation and finds its path to it.

Could probably be solved by shortening the the last vector of the path found by a little bit.

**Comment by merloechoed** [comment]
Removing the "LM" from the title, as I think it affects all companions, as pets are basically the same behind the scenes both for lore-master and captain.

**Comment by woodx319** [comment]
Thinking is quite different from knowing, and captain pets do not fly.

**Comment by merloechoed** [comment]
Just tested with bear, same thing happens.

Looks like it really is a path-finding issue, specially if the target is standing still.

![2023-01-13_20-46-14](https://user-images.githubusercontent.com/122577244/212438293-46098cdf-0d8c-42b3-8d16-c0960bd32855.jpg)

**Comment by woodx319** [comment]
good find, and you've noticed more likely if the target is stationary? good to know as well nice work man

**Comment by merloechoed** [comment]
Yes, happens way more often if the target is stationary, even more if the target is attacked from behind and has to rotate before moving towards the player's pet.

(Also the flying from pets and other creatures are most likely an animation illusion, they are probably just moving on the ground but animated as flying).
